Tasty test case TestLogarithms failed

## Reproduction steps

Scala version: scala/scala 2.13.x branch

```
sbt:root> restarrFull
sbt:root> compile
sbt:root> tasty/test
```

## Problem
Test case tastytest.TestLogarithms failed.
```
ERROR: tastytest.TestLogarithms failed: AssertionError 1 error at iteration 0:
AssertionError in TestLogarithms.`test`: assertion failed in 
  tastytest.Suite.runImpl(Suite.scala:59)
  tastytest.Suite.main(Suite.scala:40)
  tastytest.TestLogarithms.main(TestLogarithms.scala)
  java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
  java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  java.base/java.lang.reflect.Method.invoke(Method.java:566)
  scala.tools.tastytest.internal.Runner$.$anonfun$run$2(Runner.scala:26)
  scala.util.DynamicVariable.withValue(DynamicVariable.scala:59)
  scala.Console$.withErr(Console.scala:193)
  ....................
  com.novocode.junit.JUnitTask.execute(JUnitTask.java:64)
  sbt.ForkMain$Run.lambda$runTest$1(ForkMain.java:413)
  java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
  java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
  java.base/java.lang.Thread.run(Thread.java:829)
[error] Test scala.tools.tastytest.TastyTestJUnit.run failed: java.lang.AssertionError: scala.tools.tastytest.TestFailure: 1 error. Fix tastytest.TestLogarithms., took 3.747 sec

```
My environment:
MacOS M1 platform. 
openjdk version "11.0.18" 2023-01-17
OpenJDK Runtime Environment Homebrew (build 11.0.18+0)
OpenJDK 64-Bit Server VM Homebrew (build 11.0.18+0, mixed mode)

I have done some investigation. The following codes have unexpected result:
```
scala> math.exp(math.log(math.exp(math.log(2)) + math.exp(math.log(3))))
val res0: Double = 5.0

```
In another platform(xx86_64), the result is:
```
scala> math.exp(math.log(math.exp(math.log(2)) + math.exp(math.log(3))))
val res0: Double = 4.999999999999999
```

I think it is M1 arm64 platform issue. Need to try it on other arm64 platform.

--- a/test/tasty/run/src-2/tastytest/TestLogarithms.scala
+++ b/test/tasty/run/src-2/tastytest/TestLogarithms.scala
@@ -4,10 +4,10 @@ import Logarithms._
 
 object TestLogarithms extends Suite("TestLogarithms") {
 
-  val Some(l1) = Logarithm.of(2)
-  val Some(l2) = Logarithm.of(3)
+  val Some(l1) = Logarithm.of(10)
+  val Some(l2) = Logarithm.of(100)
 
-  test(assert((l1 + l2).toDouble == 4.999999999999999))
-  test(assert((l1 * l2).toDouble == 6.0))
+  test(assert((l1 + l2).toDouble == 109.99999999999997))
+  test(assert((l1 * l2).toDouble == 1000.0))
 
 }

--- a/test/tasty/run/src-3/tastytest/Logarithms.scala
+++ b/test/tasty/run/src-3/tastytest/Logarithms.scala
@@ -1,5 +1,7 @@
 package tastytest
 
+import java.lang.StrictMath
+
 object Logarithms {
 
   opaque type Logarithm = Double
@@ -8,16 +10,16 @@ object Logarithms {
 
     // These are the two ways to lift to the Logarithm type
 
-    private[Logarithms] def apply(d: Double): Logarithm = math.log(d)
+    private[Logarithms] def apply(d: Double): Logarithm = StrictMath.log10(d)
 
     def of(d: Double): Option[Logarithm] =
-      if (d > 0.0) Some(math.log(d)) else None
+      if (d > 0.0) Some(StrictMath.log10(d)) else None
   }
 
   // implicit define cross compatible public APIs for opaque types
   final implicit class LogarithmOps(val logarithm: Logarithm) extends AnyVal {
-    def toDouble: Double = math.exp(logarithm)
-    def + (other: Logarithm): Logarithm = Logarithm(math.exp(logarithm) + math.exp(other))
+    def toDouble: Double = StrictMath.pow(10.0, logarithm)
+    def + (other: Logarithm): Logarithm = Logarithm(StrictMath.pow(10.0, logarithm) + StrictMath.pow(10.0, other))
     def * (other: Logarithm): Logarithm = logarithm + other
   }
 

